src/test/mir-opt/remove-never-const.rs

```diff
@@ -1,23 +0,0 @@
-// This was originally a regression test for #66975 - ensure that we do not generate never typed
-// consts in codegen. We also have tests for this that catches the error, see
-// src/test/ui/consts/const-eval/index-out-of-bounds-never-type.rs.
-
-// Force generation of optimized mir for functions that do not reach codegen.
-// compile-flags: --emit mir,link
-
-#![feature(const_panic)]
-#![feature(never_type)]
-#![warn(const_err)]
-
-struct PrintName<T>(T);
-
-impl<T> PrintName<T> {
-    const VOID: ! = panic!();
-}
-
-// EMIT_MIR remove_never_const.no_codegen.PreCodegen.after.mir
```

Comment:
why was this test removed? Does it error now?

Reply:
When compiling with `--emit mir` it now fails with:

```
error[E0080]: any use of this value will cause an error
  --> src/test/mir-opt/remove-never-const.rs:15:21
   |
15 |     const VOID: ! = panic!();
   |     ----------------^^^^^^^^-
   |                     |
   |                     the evaluated program panicked at 'explicit panic', src/test/mir-opt/remove-never-const.rs:15:21
```